instant-segment-py/src/lib.rs

```diff
@@ -31,23 +32,29 @@ impl Segmenter {
     /// The `unigrams` iterator should yield `(str, float)` items, while the `bigrams`
     /// iterator should yield `((str, str), float)` items.
     #[new]
-    fn new(unigrams: &PyIterator, bigrams: &PyIterator) -> PyResult<Self> {
+    fn new(unigrams: &Bound<'_, PyIterator>, bigrams: &Bound<'_, PyIterator>) -> PyResult<Self> {
         let unigrams = unigrams
-            .map(|item| {
-                let item = item?;
-                let key = item.get_item(0)?.extract::<&str>()?;
-                let val = item.get_item(1)?.extract::<f64>()?;
+            .iter()?
+            .map(|result| {
+                let item = result?;
+                let key = item.get_item(0)?;
+                let key = key.extract::<&str>()?;
+                let val = item.get_item(1)?;
+                let val = val.extract::<f64>()?;
```

Comment:
nit: I think I preferred the old version of this but that's pretty subjective of course

Reply:
Me too, but it's not allowed anymore -- need to capture the borrowed data on the stack until the end of the function.